From 5f08cd2208c8a6edcbf7be6511089d8a600d90d3 Mon Sep 17 00:00:00 2001
From: Keith Maxwell <keith.maxwell@gmail.com>
Date: Wed, 15 Apr 2020 10:30:31 +0100
Subject: [PATCH] fix(deps): use mock from stdlib in tests

This change removes an unnecessary dependency in the tests.

The [mock package] is a backport of functionality from the Python 3.3 standard library to Python 2.7. google-auth-library-python is [not supported] on
Python 2.7. Using `import mock` instead of `from unittest import mock` adds an unnecessary dependency, for no apparent gain.

[mock package]: https://pypi.org/project/mock/
[not supported]: https://github.com/googleapis/google-auth-library-python#deprecated-python-versions
---
 tests/compute_engine/test__metadata.py   | 2 +-
 tests/compute_engine/test_credentials.py | 2 +-
 tests/conftest.py                        | 2 +-
 tests/crypt/test__python_rsa.py          | 2 +-
 tests/oauth2/test__client.py             | 2 +-
 tests/oauth2/test_credentials.py         | 2 +-
 tests/oauth2/test_id_token.py            | 2 +-
 tests/oauth2/test_service_account.py     | 2 +-
 tests/test__cloud_sdk.py                 | 2 +-
 tests/test__default.py                   | 2 +-
 tests/test__oauth2client.py              | 2 +-
 tests/test_app_engine.py                 | 2 +-
 tests/test_iam.py                        | 2 +-
 tests/test_impersonated_credentials.py   | 2 +-
 tests/test_jwt.py                        | 2 +-
 tests/transport/test__mtls_helper.py     | 2 +-
 tests/transport/test_grpc.py             | 2 +-
 tests/transport/test_mtls.py             | 2 +-
 tests/transport/test_requests.py         | 2 +-
 tests/transport/test_urllib3.py          | 2 +-
 20 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/tests/compute_engine/test__metadata.py b/tests/compute_engine/test__metadata.py
index 8b5eece7..c376dbf3 100644
--- a/tests/compute_engine/test__metadata.py
+++ b/tests/compute_engine/test__metadata.py
@@ -16,7 +16,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 from six.moves import http_client
 from six.moves import reload_module
diff --git a/tests/compute_engine/test_credentials.py b/tests/compute_engine/test_credentials.py
index 98def0fc..486eb618 100644
--- a/tests/compute_engine/test_credentials.py
+++ b/tests/compute_engine/test_credentials.py
@@ -14,7 +14,7 @@
 import base64
 import datetime
 
-import mock
+from unittest import mock
 import pytest
 import responses
 
diff --git a/tests/conftest.py b/tests/conftest.py
index cf8a0f9e..7b5eb1d0 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -15,7 +15,7 @@
 import os
 import sys
 
-import mock
+from unittest import mock
 import pytest
 
 
diff --git a/tests/crypt/test__python_rsa.py b/tests/crypt/test__python_rsa.py
index 886ee55a..f2d31159 100644
--- a/tests/crypt/test__python_rsa.py
+++ b/tests/crypt/test__python_rsa.py
@@ -15,7 +15,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 from pyasn1_modules import pem
 import pytest
 import rsa
diff --git a/tests/oauth2/test__client.py b/tests/oauth2/test__client.py
index c3ae2af9..f2c22ab3 100644
--- a/tests/oauth2/test__client.py
+++ b/tests/oauth2/test__client.py
@@ -16,7 +16,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 import six
 from six.moves import http_client
diff --git a/tests/oauth2/test_credentials.py b/tests/oauth2/test_credentials.py
index 76aa463c..93126e10 100644
--- a/tests/oauth2/test_credentials.py
+++ b/tests/oauth2/test_credentials.py
@@ -18,7 +18,7 @@
 import pickle
 import sys
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _helpers
diff --git a/tests/oauth2/test_id_token.py b/tests/oauth2/test_id_token.py
index ff858078..28750cc3 100644
--- a/tests/oauth2/test_id_token.py
+++ b/tests/oauth2/test_id_token.py
@@ -15,7 +15,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import environment_vars
diff --git a/tests/oauth2/test_service_account.py b/tests/oauth2/test_service_account.py
index 897374a6..70460bcd 100644
--- a/tests/oauth2/test_service_account.py
+++ b/tests/oauth2/test_service_account.py
@@ -16,7 +16,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 
 from google.auth import _helpers
 from google.auth import crypt
diff --git a/tests/test__cloud_sdk.py b/tests/test__cloud_sdk.py
index 33776042..f8063fd2 100644
--- a/tests/test__cloud_sdk.py
+++ b/tests/test__cloud_sdk.py
@@ -17,7 +17,7 @@
 import os
 import subprocess
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _cloud_sdk
diff --git a/tests/test__default.py b/tests/test__default.py
index 35000b04..8ef8b6cb 100644
--- a/tests/test__default.py
+++ b/tests/test__default.py
@@ -15,7 +15,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _default
diff --git a/tests/test__oauth2client.py b/tests/test__oauth2client.py
index 6b1112b5..9002a40c 100644
--- a/tests/test__oauth2client.py
+++ b/tests/test__oauth2client.py
@@ -16,7 +16,7 @@
 import os
 import sys
 
-import mock
+from unittest import mock
 import oauth2client.client
 import oauth2client.contrib.gce
 import oauth2client.service_account
diff --git a/tests/test_app_engine.py b/tests/test_app_engine.py
index 9dfdfa65..e363bad3 100644
--- a/tests/test_app_engine.py
+++ b/tests/test_app_engine.py
@@ -14,7 +14,7 @@
 
 import datetime
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import app_engine
diff --git a/tests/test_iam.py b/tests/test_iam.py
index c98a138f..95555925 100644
--- a/tests/test_iam.py
+++ b/tests/test_iam.py
@@ -16,7 +16,7 @@
 import datetime
 import json
 
-import mock
+from unittest import mock
 import pytest
 from six.moves import http_client
 
diff --git a/tests/test_impersonated_credentials.py b/tests/test_impersonated_credentials.py
index 19e2f342..53374704 100644
--- a/tests/test_impersonated_credentials.py
+++ b/tests/test_impersonated_credentials.py
@@ -16,7 +16,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 from six.moves import http_client
 
diff --git a/tests/test_jwt.py b/tests/test_jwt.py
index 488aee46..5a8cb501 100644
--- a/tests/test_jwt.py
+++ b/tests/test_jwt.py
@@ -17,7 +17,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _helpers
diff --git a/tests/transport/test__mtls_helper.py b/tests/transport/test__mtls_helper.py
index 04d0b56d..bfbabc18 100644
--- a/tests/transport/test__mtls_helper.py
+++ b/tests/transport/test__mtls_helper.py
@@ -15,7 +15,7 @@
 import os
 import re
 
-import mock
+from unittest import mock
 from OpenSSL import crypto
 import pytest
 
diff --git a/tests/transport/test_grpc.py b/tests/transport/test_grpc.py
index c3da76df..9f073dfc 100644
--- a/tests/transport/test_grpc.py
+++ b/tests/transport/test_grpc.py
@@ -16,7 +16,7 @@
 import os
 import time
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _helpers
diff --git a/tests/transport/test_mtls.py b/tests/transport/test_mtls.py
index d3bc3915..4c034123 100644
--- a/tests/transport/test_mtls.py
+++ b/tests/transport/test_mtls.py
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import exceptions
diff --git a/tests/transport/test_requests.py b/tests/transport/test_requests.py
index 77e1527a..8c4f33a4 100644
--- a/tests/transport/test_requests.py
+++ b/tests/transport/test_requests.py
@@ -17,7 +17,7 @@
 import sys
 
 import freezegun
-import mock
+from unittest import mock
 import OpenSSL
 import pytest
 import requests
diff --git a/tests/transport/test_urllib3.py b/tests/transport/test_urllib3.py
index 1a1c0a1e..929570e9 100644
--- a/tests/transport/test_urllib3.py
+++ b/tests/transport/test_urllib3.py
@@ -14,7 +14,7 @@
 
 import sys
 
-import mock
+from unittest import mock
 import OpenSSL
 import pytest
 from six.moves import http_client
